apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -73,11 +71,15 @@ export class RunJob {
 
       await this.storageHelperService.getAttachments(job.payload?.attachments);
 
-      const template = await this.getNotificationTemplate({
-        _id: job._templateId,
-        environmentId: job._environmentId,
+      const notification = await this.notificationRepository.findOne({
+        _id: job._notificationId,
+        _environmentId: job._environmentId,
       });
 
+      if (!notification) {
+        throw new PlatformException(`Notification with id ${job._notificationId} not found`);
+      }
```

Comment:
We throw an error here because it's an exceptional case for a job to exist without a corresponding notification.